sdk/cosmos/azure-cosmos/azure/cosmos/diagnostics.py

```diff
@@ -84,3 +89,10 @@ def __getattr__(self, name):
         if key in self._common:
             return self._headers[key]
         raise AttributeError(name)
+
+
+# def __getattr__(name):
+#     warnings.warn("The `diagnostics.py` file is deprecated and will be removed in the future. For logging " +
+#                   "diagnostics information for the SDK, please use our CosmosHttpLoggingPolicy. For more " +
+#                   "information on this, please see our README.",
+#                   DeprecationWarning)
```

Comment:
I think you can rename `RecordDiagnostics` to private class `_RecordDiagnostics`, then update the `getattr` method to something like this:
```python
def __getattr__(name):
    if name == 'RecordDiagnostics':
        warnings.warn(
            "RecordDiagnostics is deprecated and should not be used. " +
            "For logging diagnostics information for the SDK, please use our CosmosHttpLoggingPolicy. " +
            "For more information on this, please see our README.",
            DeprecationWarning
        )
        return _RecordDiagnostics

    raise AttributeError(f"module 'azure.cosmos.diagnostics' has no attribute {name}")
```
That should make the class still accessible, but will no longer appear in the docs.